packages/pn-personafisica-login/src/App.tsx

```diff
@@ -2,29 +2,40 @@ import { useEffect, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import { Box } from '@mui/material';
-import { initLocalization, useMultiEvent, useTracking } from '@pagopa-pn/pn-commons';
+import {
+  getSessionLanguage,
+  initLocalization,
+  useMultiEvent,
+  useTracking,
+} from '@pagopa-pn/pn-commons';
 
 import Router from './navigation/routes';
 import { getConfiguration } from './services/configuration.service';
 import './utility/onetrust';
 
 const App = () => {
   const configuration = useMemo(() => getConfiguration(), []);
-  
+
   useTracking(configuration.MIXPANEL_TOKEN, process.env.NODE_ENV);
 
   const [clickVersion] = useMultiEvent({
     callback: () => console.log(`v${configuration.VERSION}`),
   });
 
-  const { t } = useTranslation(['common']);
+  const { t, i18n } = useTranslation(['common']);
   const [isInitialized, setIsInitialized] = useState(false);
 
+  const handleSetUserLanguage = async () => {
+    const language = getSessionLanguage() || 'it';
+    await i18n.changeLanguage(language);
```

Comment:
This can lead to an extra render. We can try to create an i18next language detector plugin https://www.i18next.com/misc/creating-own-plugins.

If it'll work, we can do the same thing for all the apps